src/function/scalar/list/list_reverse.cpp

```diff
@@ -0,0 +1,115 @@
+#include "duckdb/common/types/data_chunk.hpp"
+#include "duckdb/function/scalar/nested_functions.hpp"
+#include "duckdb/planner/expression/bound_function_expression.hpp"
+#include "duckdb/planner/expression/bound_parameter_expression.hpp"
+#include "duckdb/planner/expression_binder.hpp"
+
+namespace duckdb {
+
+static void ListReverseFunction(DataChunk &args, ExpressionState &state, Vector &result) {
+	D_ASSERT(args.ColumnCount() == 1);
+	auto count = args.size();
+
+	Vector &input_list = args.data[0];
+	if (input_list.GetType().id() == LogicalTypeId::SQLNULL) {
+		result.SetVectorType(VectorType::CONSTANT_VECTOR);
+		ConstantVector::SetNull(result, true);
+		return;
+	}
+
+	// get the input list data
+	UnifiedVectorFormat input_list_data;
+	input_list.ToUnifiedFormat(count, input_list_data);
+	auto input_list_entries = UnifiedVectorFormat::GetData<list_entry_t>(input_list_data);
+
+	// get the result entires
+	auto result_entries = FlatVector::GetData<list_entry_t>(result);
+
+	// get the validity mask of the result to avoid complications later
+	auto &result_validity = FlatVector::Validity(result);
+
+	// get the input child vector to slice it later
+	auto &input_list_child = ListVector::GetEntry(input_list);
+
+	// create a selection vector for slicing the child vector
+	SelectionVector rev_sel(ListVector::GetListSize(input_list));
+
+	for (idx_t i = 0; i < count; i++) {
+		auto input_list_list_index = input_list_data.sel->get_index(i);
+
+		// check if the input list is valid
+		if (!input_list_data.validity.RowIsValid(input_list_list_index)) {
+			result_validity.SetInvalid(i);
+			continue;
+		};
+
+		D_ASSERT(input_list_data.validity.RowIsValid(input_list_list_index));
+		const auto &input_list_entry = input_list_entries[input_list_list_index];
+		result_entries[i].length += input_list_entry.length;
```

Comment:
These entries are likely uninitialized, and as stated before, this step is just completely not necessary as both the length and the offsets for each list will be unchanged from the input